Detectors/MUON/MCH/Simulation/src/Digitizer.cxx

```diff
@@ -157,12 +160,53 @@ int Digitizer::processHit(const Hit& hit, int detID, double event_time)
       q *= chargenon;
     }
     auto signal = resp.response(q);
-    digits.emplace_back(padid, signal);
+    digits.emplace_back(time, padid, signal);
+    multimap.emplace(padid, digits.size());
     ++ndigits;
   });
   return ndigits;
 }
+//______________________________________________________________________
+void Digitizer::mergeDigits(std::vector<Digit>& digits){
```

Comment:
here I guess there are many ways to implement this. The choice should be performance- and readability- driven. 

That's one instance where we'd greatly benefit for a narrow unit test (well, unit benchmark in that case).
For instance generating fake digits with e.g. a known fraction of duplicates (fraction can then be varied from 0 to 1 to see how the implementation behaves), and then timing the merging method(s). 

Because as it is my guts tell me it's not efficient to use a multi map, a set and removal from a vector all at once. But my guts might well be totally wrong. Only measuring can tell.